File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.js
/**
 * @param {number[]} nums
 * @param {number} minK
 * @param {number} maxK
 * @return {number}
 */
var countSubarrays = function(nums, minK, maxK) {
  let [ans, lo, hi, bad] = [0, -1, -1, -1];
  
  nums.forEach((num, i) => {
    if (num < minK || num > maxK) bad = i;
    if (num === minK) lo = i;
    if (num === maxK) hi = i;
    
    ans += Math.max(0, Math.min(lo, hi) - bad);
  })
  
  return ans;
};

File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.py
class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        ans, bad, lo, hi = 0, -1, -1, -1
        for i, num in enumerate(nums):
            if num < minK or num > maxK:
                bad = i
            if num == minK:
                lo = i
            if num == maxK:
                hi = i
            ans += max(0, min(lo, hi) - bad)
        return ans